SIGSEV when receving message with a quote

Hello!

When receving message with a quote (for instance, as part of a reply to another message) I receive the following stacktrace in my bot logs:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x6a91b1]

goroutine 404 [running]:
github.com/chatmail/rpc-client-go/v2/deltachat.unmarshalMessageQuote({0x1e8d8af82000, 0x2af, 0x2c0}, 0x0)
 /go/pkg/mod/github.com/chatmail/rpc-client-go/v2@v2.49.0/deltachat/types.go:2307 +0x171
github.com/chatmail/rpc-client-go/v2/deltachat.(*Message).UnmarshalJSON(0x1e8d8b126000, {0x1e8d8ae55800, 0x772, 0x800})
 /go/pkg/mod/github.com/chatmail/rpc-client-go/v2@v2.49.0/deltachat/types.go:2012 +0x4bd
encoding/json.(*decodeState).object(0x1e8d8adce2d0, {0xb868c0?, 0x1e8d8acd2430?, 0x0?})
 /usr/local/go/src/encoding/json/decode.go:610 +0x68e
 encoding/json.(*decodeState).value(0x1e8d8adce2d0, {0xb868c0?, 0x1e8d8acd2430?, 0x4818a5?})
 /usr/local/go/src/encoding/json/decode.go:380 +0x3e
encoding/json.(*decodeState).unmarshal(0x1e8d8adce2d0, {0xb868c0?, 0x1e8d8acd2430?})
 /usr/local/go/src/encoding/json/decode.go:183 +0x11e
encoding/json.Unmarshal({0x1e8d8ae55800, 0x772, 0x800}, {0xb868c0, 0x1e8d8acd2430})
 /usr/local/go/src/encoding/json/decode.go:113 +0xf9
github.com/creachadair/jrpc2.(*Response).UnmarshalResult(0x1e8d8acca380?, {0xb868c0?, 0x1e8d8acd2430?})
 /go/pkg/mod/github.com/creachadair/jrpc2@v1.3.5/base.go:157 +0xef
github.com/creachadair/jrpc2.(*Client).CallResult(0x1e8d8acdc160?, {0xd1b970?, 0x1380080?}, {0xcc5a91?, 0x1e8d8ad74928?}, {0xb93840?, 0x1e8d8acb85b8?}, {0xb868c0, 0x1e8d8acd2430})
 /go/pkg/mod/github.com/creachadair/jrpc2@v1.3.5/client.go:319 +0x66
github.com/chatmail/rpc-client-go/v2/deltachat.(*IOTransport).CallResult(0x1e8d8aec8620, {0xd1b970, 0x1380080}, {0xbba820, 0x1e8d8b126000}, {0xcc5a91, 0xb}, {0x1e8d8acdc160, 0x2, 0x2})
 /go/pkg/mod/github.com/chatmail/rpc-client-go/v2@v2.49.0/deltachat/io_transport.go:91 +0xe5
github.com/chatmail/rpc-client-go/v2/deltachat.(*Rpc).GetMessage(_, _, _)
 /go/pkg/mod/github.com/chatmail/rpc-client-go/v2@v2.49.0/deltachat/rpc.go:977 +0x143
.... [stack trace continues to my bot source code]
```

I've checked and this happens after bumping the local version of `rpc-client-go` go library without touching any of my code. The rust binary is in sync with the go library.

@Polpetta 

reproduced this on `github.com/chatmail/rpc-client-go/v2@v2.49.0` without a deltachat rpc  server or bot code involved. a minimal `encoding/json.Unmarshal` into `deltachat.Message` is enough when the payload contains a non-null `quote`.

minimal poc:

````
package poc_test

import (
	"encoding/json"
	"testing"

	"github.com/chatmail/rpc-client-go/v2/deltachat"
)

func TestMessageUnmarshalQuotedMessageDoesNotPanic(t *testing.T) {
	payload := []byte(`{"id":1,"quote":{"kind":"JustText","text":"quoted text"}}`)
	var msg deltachat.Message

	if err := json.Unmarshal(payload, &msg); err != nil {
		t.Fatalf("json.Unmarshal quoted message: %v", err)
	}

	if msg.Quote == nil {
		t.Fatal("quote was not decoded")
	}
}
````

run:
`go test -run TestMessageUnmarshalQuotedMessageDoesNotPanic -count=1`

observed on `v.2.49.0`

````
panic: runtime error: invalid memory address or nil pointer dereference

github.com/chatmail/rpc-client-go/v2/deltachat.unmarshalMessageQuote(..., 0x0)
	.../github.com/chatmail/rpc-client-go/v2@v2.49.0/deltachat/types.go:2301
github.com/chatmail/rpc-client-go/v2/deltachat.(*Message).UnmarshalJSON(...)
	.../github.com/chatmail/rpc-client-go/v2@v2.49.0/deltachat/types.go:2012
encoding/json.Unmarshal(...)
````

this looks like generated-client code passing a nil output pointer.

````
if len(raw.Quote) > 0 && string(raw.Quote) != "null" {
	if err := unmarshalMessageQuote(raw.Quote, s.Quote); err != nil {
		return err
	}
}
````
then unmarshalMessageQuote dereferences that nil pointer:

`*out = &v`

a minimal api-preserving fix might be:

````
if len(raw.Quote) > 0 && string(raw.Quote) != "null" {
	var quote MessageQuote
	if err := unmarshalMessageQuote(raw.Quote, &quote); err != nil {
		return err
	}
	s.Quote = &quote
}
````

i hit this from a downstream tui i'm developing when loading a chat containing a replied-to(quoted) message. i added a downstream recovery guard for now, but the standalone repro above seems to show the issue is in `Message.UnmarshalJSON ` itself.


cc: @adbenitez

--- a/v2/deltachat/rpc_test.go
+++ b/v2/deltachat/rpc_test.go
@@ -974,6 +974,25 @@ func TestMsg_GetMessageHtml(t *testing.T) {
 	})
 }
 
+func TestMsg_GetMessage(t *testing.T) {
+	t.Parallel()
+	acfactory.WithGroup(func(rpc *Rpc, accId uint32, chatId uint32) {
+		// message without quote
+		msgId1, err := rpc.SendMsg(accId, chatId, MessageData{Text: strptr("hello")})
+		require.Nil(t, err)
+		msg1, err := rpc.GetMessage(accId, msgId1)
+		require.Nil(t, err)
+		require.Nil(t, msg1.Quote)
+
+		// message with quote
+		msgId2, err := rpc.SendMsg(accId, chatId, MessageData{Text: strptr("hi"), QuotedMessageId: &msgId1})
+		require.Nil(t, err)
+		msg2, err := rpc.GetMessage(accId, msgId2)
+		require.Nil(t, err)
+		require.NotNil(t, msg2.Quote)
+	})
+}
+
 func TestMsg_GetMessageNotificationInfo(t *testing.T) {
 	t.Parallel()
 	acfactory.WithGroup(func(rpc *Rpc, accId uint32, chatId uint32) {

--- a/v2/deltachat/types.go
+++ b/v2/deltachat/types.go
@@ -2009,9 +2009,11 @@ func (s *Message) UnmarshalJSON(data []byte) error {
 	s.ViewType = raw.ViewType
 	s.WebxdcHref = raw.WebxdcHref
 	if len(raw.Quote) > 0 && string(raw.Quote) != "null" {
-		if err := unmarshalMessageQuote(raw.Quote, s.Quote); err != nil {
+		var obj MessageQuote
+		if err := unmarshalMessageQuote(raw.Quote, &obj); err != nil {
 			return err
 		}
+		s.Quote = &obj
 	}
 	return nil
 }

